WooCommerce/Classes/POS/Card Present Payments/CardPresentPaymentService.swift

```diff
@@ -27,7 +26,8 @@ final class CardPresentPaymentService: CardPresentPaymentFacade {
 
     private var paymentTask: Task<CardPresentPaymentAdaptedCollectOrderPaymentResult, Error>?
 
-    init(siteID: Int64) {
+    @MainActor
```

Comment:
Just curious: Since all methods in this class are currently marked as `@MainActor` should we mark the whole class as `@MainActor` instead? It would require to update the `disconnectReader` and `cancelPayment` functions to return `async throws` (in the protocol as well).

Reply:
I'm not sure that the whole thing needs to be, it would be nice to specialise the individual tasks which actually need to be on the MainActor if we can. Having it repeated kinda reminds about that... but yes, if we do need all the functions to be MainActor, we can do it to the class.